src/Components/Search/SearchInput.js

```diff
@@ -0,0 +1,104 @@
+import React from "react";
+import { connect } from "react-redux";
+import { SearchStyled, InputStyled, ButtonStyled } from "./SearchInput.styled";
+
+class SearchInput extends React.Component {
+  handleSearch = () => {
+    const {
+      onRequestSearchGifs,
+      fromFavorites,
+      searchInput,
+      searchFavorite,
+      onFilterFavorites
+    } = this.props;
+    if (!fromFavorites) {
+      onRequestSearchGifs(searchInput);
+    } else {
+      onFilterFavorites(searchFavorite);
+    }
+  };
+
+  handleKeyPress = e => {
+    var keyCode = e.which || e.keyCode;
+    const {
+      onRequestSearchGifs,
+      fromFavorites,
+      searchInput,
+      searchFavorite,
+      onFilterFavorites
+    } = this.props;
+
+    if (
+      keyCode === 13 ||
```

Comment:
As a tip, always use `const` for these kind of cases:

`const ENTER_KEY_CODE = 13;`